Rover/sailboat.cpp

```diff
@@ -260,6 +264,58 @@ void Sailboat::get_throttle_and_mainsail_out(float desired_speed, float &throttl
         wingsail_out *= -1.0f;
     }
 
+    //
+    // direct mast rotation control
+    //
+
+    if (!is_positive(desired_speed)) {
+        // rotating sails can be used to reverse, but not in this version
+        mast_rotation_out = 0.0f;
+    } else {
+        // + is wind over starboard side, - is wind over port side
+        float wind_dir_apparent_signed = degrees(rover.g2.windvane.get_apparent_wind_direction_rad());
+        float wind_dir_apparent_abs = fabsf(wind_dir_apparent_signed);
+        float wind_dir_apparent_sign = is_equal(wind_dir_apparent_abs, 0.0f) ? 0.0f : wind_dir_apparent_signed / wind_dir_apparent_abs;
+
+        // compute absolute sail rotation
+        float mast_rotation_angle = 0.0f;
+        bool mast_rotation_necessary = true;
+        if (wind_dir_apparent_abs < sail_angle_ideal) {
+            // in irons, center the sail.
+            mast_rotation_angle = 0.0f;
+        } else if (wind_dir_apparent_abs < (90.0f + sail_angle_ideal)) {
+            // use sail as a lift device, at ideal angle of attack
+            // (but depower to prevent excessive heel)
+            float power_coefficient = 1.0f - pid_offset/100.0f;  // pid_offset==0.0f: no restrictions on power; ==100.0f: depower entirely
```

Comment:
```suggestion
            float power_coefficient = 1.0f - pid_offset * 0.01f;  // pid_offset==0.0f: no restrictions on power; ==100.0f: depower entirely
```
if your going to do another patch, you might as well swap out for a multiply. The comment would probably also be better up where we first get the PID offset.

Reply:
I was going back and forth on that. In the end I decided that I was the odd man out and left this math in my piece of the code. If I were going to move this up to the initial calculation, I'd do two things:

1. Do all the constraining in `get_sail_out_from_heel()` instead of some there and some in `sailboat.cpp`
2. Make `get_sail_out_from_heel() return a value in range [0..1] instead of [0..100] and multiply it in mainsail_out and wingsail_out calculations.

I didn't do it to avoid disturbing code that has its own flow to it, and which I cannot test, having no access to boats with relevant configurations.

I'm happy to do it though if you prefer.

The division is probably no more expensive (gcc can normally figure this out when it's a constant). I used it for symmetry with multiplication by 100.0f nu line 223. Happy to change as part of the bigger change above if you want it.

Reply:
The comment could go on `get_sail_out_from_heel()` and there would be no need for it in `sailboat.cpp`
I basically don't like upsetting other people's code unless they ask me to.

Reply:
I just pushed ec8815dd24636e0f4b2c0c5fc149d76b53068064 which addresses this idea. If you think it's an improvement, I can squash it into one commit. If you think it's uglier than existing code, I'm happy to take it off.